File: array_and_strings/find_peak_element.py
from math import inf

class Solution:
    def findPeakElement(self, nums):
        start, end = 1, len(nums)
        nums = [-inf] + nums + [-inf]
        mid = end
        while start < end:
            if nums[mid] < nums[mid + 1]:
                start = mid + 1
            else:
                end = mid
            mid = (end + start) // 2
        return mid - 1

File: array_and_strings/find_peak_element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
    let [start, end] = [1, nums.length]
    let mid = end;
    nums.unshift(-Infinity);
    nums.push(-Infinity);
    while (start < end) {
        if (nums[end] > nums[mid]) {
            start = mid + 1;
        } else {
            end = mid;
        }
        mid = parseInt((end + start) / 2); 
    }
    return mid - 1;
}